tests/runtime/engine/utils.py

```diff
@@ -99,6 +100,13 @@ def run_test(test):
 
             if test.before:
                 before = subprocess.Popen(test.before, shell=True, preexec_fn=os.setsid)
+                waited=0
+                with open(devnull, 'w') as dn:
+                    while subprocess.call("pidof $(basename $(echo %s | awk '{print $NF}'))" % test.before, shell=True, stdout=dn, stderr=dn) != 0:
```

Comment:
` $(basename $(echo %s | awk '{print $NF}'`

might be nicer to do that basename thing in python, that way you can avoid forking yet another shell just to run pidof.

Reply:
I reworked it in python, I also added a note on the limitations of this heuristic. It fits the current cases by just taking the basename of the last shell token.

If we augment tests with wrappers like mountns_wrapper rather than pass arguments, this approach works. If future tests take arguments, this will need to change.

I think the established pattern though is to wrap a binary or just make a separate one to keep them simple and separate, rather than having to do argument parsing.